error[E0277]: the trait bound `Q: FromParam<'_>` is not satisfied
 --> tests/ui-fail/route-type-errors.rs:8:7
  |
8 | fn f0(foo: Q) {} //~ ERROR FromParam
  |       ^^^^^^ the trait `FromParam<'_>` is not implemented for `Q`
  |
  = help: the following other types implement trait `FromParam<'a>`:
            &'a RawStr
            Cow<'a, str>
            IpAddr
            Ipv4Addr
            Ipv6Addr
            NonZeroI128
            NonZeroI16
            NonZeroI32
          and 30 others

error[E0277]: the trait bound `Q: FromSegments<'_>` is not satisfied
  --> tests/ui-fail/route-type-errors.rs:11:7
   |
11 | fn f1(foo: Q) {} //~ ERROR FromSegments
   |       ^^^^^^ the trait `FromSegments<'_>` is not implemented for `Q`
   |
   = help: the following other types implement trait `FromSegments<'a>`:
             Option<T>
             PathBuf
             Result<T, <T as FromSegments<'a>>::Error>
             Segments<'a>

error[E0277]: the trait bound `Q: FromFormValue<'_>` is not satisfied
  --> tests/ui-fail/route-type-errors.rs:14:7
   |
14 | fn f2(foo: Q) {} //~ ERROR FromFormValue
   |       ^^^^^^ the trait `FromFormValue<'_>` is not implemented for `Q`
   |
   = help: the following other types implement trait `FromFormValue<'v>`:
             &'v RawStr
             IpAddr
             Ipv4Addr
             Ipv6Addr
             NonZeroI128
             NonZeroI16
             NonZeroI32
             NonZeroI64
           and 29 others

error[E0277]: the trait bound `Q: FromQuery<'_>` is not satisfied
  --> tests/ui-fail/route-type-errors.rs:17:7
   |
17 | fn f3(foo: Q) {} //~ ERROR FromQuery
   |       ^^^^^^ the trait `FromQuery<'_>` is not implemented for `Q`
   |
   = help: the following other types implement trait `FromQuery<'q>`:
             Form<T>
             LenientForm<T>
             Option<T>
             Result<T, <T as FromQuery<'q>>::Error>

error[E0277]: the trait bound `Q: FromDataSimple` is not satisfied
  --> tests/ui-fail/route-type-errors.rs:20:7
   |
20 | fn f4(foo: Q) {} //~ ERROR FromData
   |       ^^^^^^ the trait `FromDataSimple` is not implemented for `Q`
   |
   = help: the following other types implement trait `FromDataSimple`:
             Vec<u8>
             std::string::String
   = note: required because of the requirements on the impl of `FromData<'_>` for `Q`

error[E0277]: the trait bound `Q: FromRequest<'_, '_>` is not satisfied
  --> tests/ui-fail/route-type-errors.rs:23:7
   |
23 | fn f5(a: Q, foo: Q) {}
   |       ^^^^ the trait `FromRequest<'_, '_>` is not implemented for `Q`
   |
   = help: the following other types implement trait `FromRequest<'a, 'r>`:
             &'a rocket::http::Accept
             &'a rocket::http::ContentType
             &'a rocket::http::uri::Origin<'a>
             &'r Route
             Cookies<'a>
             Flash<&'a rocket::Request<'r>>
             Option<T>
             Result<T, <T as FromRequest<'a, 'r>>::Error>
           and 3 others

error[E0277]: the trait bound `Q: FromParam<'_>` is not satisfied
  --> tests/ui-fail/route-type-errors.rs:23:13
   |
23 | fn f5(a: Q, foo: Q) {}
   |             ^^^^^^ the trait `FromParam<'_>` is not implemented for `Q`
   |
   = help: the following other types implement trait `FromParam<'a>`:
             &'a RawStr
             Cow<'a, str>
             IpAddr
             Ipv4Addr
             Ipv6Addr
             NonZeroI128
             NonZeroI16
             NonZeroI32
           and 30 others

error[E0277]: the trait bound `Q: FromRequest<'_, '_>` is not satisfied
  --> tests/ui-fail/route-type-errors.rs:28:7
   |
28 | fn f6(a: Q, foo: Q, good: usize, bar: Q) {}
   |       ^^^^ the trait `FromRequest<'_, '_>` is not implemented for `Q`
   |
   = help: the following other types implement trait `FromRequest<'a, 'r>`:
             &'a rocket::http::Accept
             &'a rocket::http::ContentType
             &'a rocket::http::uri::Origin<'a>
             &'r Route
             Cookies<'a>
             Flash<&'a rocket::Request<'r>>
             Option<T>
             Result<T, <T as FromRequest<'a, 'r>>::Error>
           and 3 others

error[E0277]: the trait bound `Q: FromParam<'_>` is not satisfied
  --> tests/ui-fail/route-type-errors.rs:28:13
   |
28 | fn f6(a: Q, foo: Q, good: usize, bar: Q) {}
   |             ^^^^^^ the trait `FromParam<'_>` is not implemented for `Q`
   |
   = help: the following other types implement trait `FromParam<'a>`:
             &'a RawStr
             Cow<'a, str>
             IpAddr
             Ipv4Addr
             Ipv6Addr
             NonZeroI128
             NonZeroI16
             NonZeroI32
           and 30 others

error[E0277]: the trait bound `Q: FromParam<'_>` is not satisfied
  --> tests/ui-fail/route-type-errors.rs:28:34
   |
28 | fn f6(a: Q, foo: Q, good: usize, bar: Q) {}
   |                                  ^^^^^^ the trait `FromParam<'_>` is not implemented for `Q`
   |
   = help: the following other types implement trait `FromParam<'a>`:
             &'a RawStr
             Cow<'a, str>
             IpAddr
             Ipv4Addr
             Ipv6Addr
             NonZeroI128
             NonZeroI16
             NonZeroI32
           and 30 others
