error: named and unnamed parameters cannot be mixed
 --> tests/ui-fail/typed-uris-invalid-syntax.rs:9:18
  |
9 |     uri!(simple: id = 100, "Hello"); //~ ERROR named and unnamed
  |                  ^^^^^^^^^^^^^^^^^

error: named and unnamed parameters cannot be mixed
  --> tests/ui-fail/typed-uris-invalid-syntax.rs:10:18
   |
10 |     uri!(simple: "Hello", id = 100); //~ ERROR named and unnamed
   |                  ^^^^^^^^^^^^^^^^^

error: expected `:`
  --> tests/ui-fail/typed-uris-invalid-syntax.rs:11:16
   |
11 |     uri!(simple,); //~ ERROR expected `:`
   |                ^

error: expected argument list after `:`
  --> tests/ui-fail/typed-uris-invalid-syntax.rs:12:16
   |
12 |     uri!(simple:); //~ ERROR argument list
   |                ^

error: unexpected end of input: expected ',' followed by route path
  --> tests/ui-fail/typed-uris-invalid-syntax.rs:13:10
   |
13 |     uri!("/mount"); //~ ERROR route path
   |          ^^^^^^^^

error: unexpected end of input, expected identifier
  --> tests/ui-fail/typed-uris-invalid-syntax.rs:14:5
   |
14 |     uri!("/mount",); //~ ERROR expected identifier
   |     ^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `uri` (in Nightly builds, run with -Z macro-backtrace for more info)

error: invalid mount point; mount points must be static, absolute URIs: `/example`
  --> tests/ui-fail/typed-uris-invalid-syntax.rs:15:10
   |
15 |     uri!("mount", simple); //~ invalid mount point
   |          ^^^^^^^

error: invalid mount point; mount points must be static, absolute URIs: `/example`
  --> tests/ui-fail/typed-uris-invalid-syntax.rs:16:10
   |
16 |     uri!("/mount/<id>", simple); //~ invalid mount point
   |          ^^^^^^^^^^^^^

error: unexpected end of input, call to `uri!` cannot be empty
  --> tests/ui-fail/typed-uris-invalid-syntax.rs:17:5
   |
17 |     uri!(); //~ unexpected end of input
   |     ^^^^^^
   |
   = note: this error originates in the macro `uri` (in Nightly builds, run with -Z macro-backtrace for more info)

error: unexpected end of input, expected expression
  --> tests/ui-fail/typed-uris-invalid-syntax.rs:18:5
   |
18 |     uri!(simple: id = ); //~ expected expression
   |     ^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `uri` (in Nightly builds, run with -Z macro-backtrace for more info)
