error: unexpected token in attribute
 --> tests/ui/auto_enum/attribute.rs:9:16
  |
9 |         #[never(foo)] //~ ERROR unexpected token
  |                ^

error: unexpected token in attribute
  --> tests/ui/auto_enum/attribute.rs:19:17
   |
19 |         #[nested(foo)] //~ ERROR unexpected token
   |                 ^

error: #[rec] has been removed and replaced with #[nested]
  --> tests/ui/auto_enum/attribute.rs:29:9
   |
29 |         #[rec] //~ ERROR #[rec] has been removed and replaced with #[nested]
   |         ^^^^^^
